File: testbed/workflows/video-processing/03-merge-videos/src/util/rest.ts
import { StructuredReturn } from 'faas-js-runtime';

export function reportInvalidS3ObjRef(): StructuredReturn {
    return {
        statusCode: 400,
        body: {
            message: 'Incoming body is not a valid S3ObjectReference.',
        },
        headers: {
            'content-type': 'application/json',
        },
    };
}

export function reportInvalidVideoCutRequest(): StructuredReturn {
    return {
        statusCode: 400,
        body: {
            message: 'Incoming body is not a valid VideoCutRequest.',
        },
        headers: {
            'content-type': 'application/json',
        },
    };
}

export function createErrorResponse(err: Error): StructuredReturn {
    return {
        statusCode: 500,
        body: {
            message: err.toString(),
        },
        headers: {
            'content-type': 'application/json',
        },
    };
}

File: testbed/workflows/face-detection/03-detect-faces/util/rest.py
from typing import Any, Mapping

def report_invalid_video_request(req_type: str) -> dict[str, Any]:
    return {
        'statusCode': 400,
        'body': {
            'message': f'Incoming body is not a valid {req_type}.',
        },
        'headers': {
            'content-type': 'application/json',
        },
    }

def report_exception(ex: Exception) -> dict[str, Any]:
    return {
        'statusCode': 500,
        'body': {
            'error': f'{ex}',
        },
        'headers': {
            'content-type': 'application/json',
        },
    }

def check_field_exists_or_error(data: Mapping[str, Any], key: str, typ: type, req_class: type):
    if type(data[key]) is not typ:
        raise Exception(f'{key} is missing from {req_class}')
